### ISSUE TITLE
v3 模型和之前模型的区别

### ISSUE THREAD

**Issue by hzwer** [hzwer/Practical-RIFE#2]
@KevenLee 不好意思我把你的问题移动到这

**Comment by KevenLee** [comment]
有个不是很明白的地方，IFBlock 里面，mask 是什么意思啊，出于什么原因考虑增加mask呢    
def forward(self, x, flow, scale=1):
        x = F.interpolate(x, scale_factor= 1. / scale, mode="bilinear", align_corners=False, recompute_scale_factor=False)
        flow = F.interpolate(flow, scale_factor= 1. / scale, mode="bilinear", align_corners=False, recompute_scale_factor=False) * 1. / scale
        feat = self.conv0(torch.cat((x, flow), 1))
        feat = self.convblock0(feat) + feat
        feat = self.convblock1(feat) + feat
        feat = self.convblock2(feat) + feat
        feat = self.convblock3(feat) + feat        
        flow = self.conv1(feat)
        mask = self.conv2(feat)
        flow = F.interpolate(flow, scale_factor=scale, mode="bilinear", align_corners=False, recompute_scale_factor=False) * scale
        mask = F.interpolate(mask, scale_factor=scale, mode="bilinear", align_corners=False, recompute_scale_factor=False)
        return flow, mask

**Comment by hzwer** [comment]
@KevenLee 前后两帧向中间 warp 的时候，会出两个结果，需要相互融合
融合权重就是 mask

**Comment by KevenLee** [comment]
> @KevenLee 前后两帧向中间 warp 的时候，会出两个结果，需要相互融合
> 融合权重就是 mask

V3比V2 明显要简洁很多，V3 里面应该没有显式地 warp，是不是说明，通过 CNN 也能可以很好地去拟合依靠光流显式地warp的操作？

**Comment by DunguTmp** [comment]
> 其实主要心得就是，psnr 不靠谱，换成 lpips 以后，随便调一下视觉效果突飞猛进

您好，想问一下。这句话的意思是说通过lpips的指标替换psnr指标来衡量最终的插帧效果还是其他的意思呢？

**Comment by hzwer** [comment]
@DunguTmp 对，最终指标看 lpips，和视觉效果还是比较一致的

**Comment by Gilgamesh666666** [comment]
您好，恭喜rife中eccv2022。我目前想在自己数据集上复现一下任意timestep的rife，请问您之后会考虑公开训练细节吗，比如4版本的rife用的什么蒸馏方法，还有跟论文上说的比，用了什么新的训练策略？因为看到4版本的rife网络结构跟论文的也不同，所以想问一下，非常感谢！

**Comment by hzwer** [comment]
@Gilgamesh666666 Hi，关于针对视觉效果的模型，参见 https://github.com/megvii-research/ECCV2022-RIFE/issues/172#issuecomment-855212336；论文结构和方法没有什么本质差别，主要是多加了一个尺度，以应对一些实际用到的高分辨率视频处理

**Comment by Gilgamesh666666** [comment]
> @Gilgamesh666666 Hi，关于针对视觉效果的模型，参见 [megvii-research/ECCV2022-RIFE#172 (comment)](https://github.com/megvii-research/ECCV2022-RIFE/issues/172#issuecomment-855212336) 论文结构和方法没有什么本质差别，主要是多加了一个尺度，以应对一些实际用到的高分辨率视频处理

您好非常感谢您的回复！看到您说3.5版本之后用了新的蒸馏方法，请问这个方便公开吗，如果不可以的话，请问用原文代码中的蒸馏方法和训练策略训练v4模型可以得到差不多的效果吗？

**Comment by hzwer** [comment]
> > @Gilgamesh666666 Hi，关于针对视觉效果的模型，参见 [megvii-research/ECCV2022-RIFE#172 (comment)](https://github.com/megvii-research/ECCV2022-RIFE/issues/172#issuecomment-855212336) 论文结构和方法没有什么本质差别，主要是多加了一个尺度，以应对一些实际用到的高分辨率视频处理
> 
> 您好非常感谢您的回复！看到您说3.5版本之后用了新的蒸馏方法，请问这个方便公开吗，如果不可以的话，请问用原文代码中的蒸馏方法和训练策略训练v4模型可以得到差不多的效果吗？

原理上的改动已经都合并进现在的代码和发表论文里
